pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -813,6 +813,141 @@ func TestUserSignupWithInvalidNameNotOK(t *testing.T) {
 	)
 }
 
+func TestUserSignupDeactivatedWithNoMUR(t *testing.T) {
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      uuid.NewV4().String(),
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			Username:    "foo#bar@redhat.com",
+			Approved:    false,
+			Deactivated: true,
+		},
+	}
+
+	r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, configMap(config.UserApprovalPolicyAutomatic), basicNSTemplateTier)
+
+	_, err := r.Reconcile(req)
+	require.NoError(t, err)
+
+	key := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      userSignup.Name,
+	}
+	instance := &v1alpha1.UserSignup{}
+	err = r.client.Get(context.TODO(), key, instance)
+	require.NoError(t, err)
+
+	test.AssertConditionsMatch(t, instance.Status.Conditions,
+		v1alpha1.Condition{
+			Type:   v1alpha1.UserSignupComplete,
+			Status: v1.ConditionTrue,
+			Reason: "Deactivated",
+		},
+	)
+}
+
+func TestUserSignupDeactivatedAfterMURCreated(t *testing.T) {
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      uuid.NewV4().String(),
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			Username: "john.doe@redhat.com",
+		},
+		Status: v1alpha1.UserSignupStatus{
+			Conditions: []v1alpha1.Condition{
+				{
+					Type:   v1alpha1.UserSignupComplete,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1alpha1.UserSignupApproved,
+					Status: v1.ConditionTrue,
+					Reason: "ApprovedAutomatically",
+				},
+			},
+			CompliantUsername: "john-doe-at-redhat-com",
+		},
+	}
+
+	mur := &v1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "john-doe-at-redhat-com",
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
+			Labels:    map[string]string{toolchainv1alpha1.MasterUserRecordUserIDLabelKey: userSignup.Name},
+		},
+	}
```

Comment:
There are some functions in toolchain-common repo that will help you with creating some of the CRs:
```
mur := murtest.NewMasterUserRecord("john-doe-at-redhat-com", murtest.MetaNamespace(operatorNamespace))
mur.Labels = map[string]string{toolchainv1alpha1.MasterUserRecordUserIDLabelKey: userSignup.Name}
```
see https://github.com/codeready-toolchain/toolchain-common/tree/master/pkg/test/masteruserrecord
it would be good if we had something similar for UserSignup too - the tests would be much shorter

Reply:
Thanks Matous, these functions make it much cleaner.